botocore/waiter.py

```diff
@@ -43,7 +45,32 @@ def create_waiter_with_client(waiter_name, waiter_model, client):
     operation_name = xform_name(single_waiter_config.operation)
     operation_method = NormalizedOperationMethod(
         getattr(client, operation_name))
-    return Waiter(
+
+    # Create a new wait method that will serve as a proxy to the underlying
+    # Waiter.wait method. This is needed to attach a docstring to the
+    # method.
+    def wait(self, **kwargs):
+        Waiter.wait(self, **kwargs)
+
+    wait.__doc__ = WaiterDocstring(
+        waiter_name=waiter_name,
+        event_emitter=client.meta.events,
+        service_model=client.meta.service_model,
+        service_waiter_model=waiter_model,
+        include_signature=False
+    )
+
+    # Rename the waiter class based on the type of waiter.
+    waiter_class_name = str('%s.Waiter.%s' % (
+        get_service_module_name(client.meta.service_model),
+        xform_name(waiter_name)))
+
+    # Create the new waiter class
+    documented_waiter_cls = type(
```

Comment:
This will create a new class object every time `client.get_waiter` is called.  Can we avoid creating new class objects?  Are we doing this only so the class name will be different?